File: src/main/java/com/wikia/tibia/repositories/NPCRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.NPCGateway;
import com.wikia.tibia.objects.NPC;

public class NPCRepository extends WikiObjectRepository {

    public NPCRepository() {
        super(NPC.class, new NPCGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/NPCRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.NPC

class NPCRepository : WikiObjectRepository<NPC>(
    wikiObjectClass = NPC::class.java,
    wikiObjectGateway = WikiObjectGateway(Contract.NPCS)
)
